test/e2e/set/scyllacluster/scyllamanager.go

```diff
@@ -207,8 +210,53 @@ var _ = g.Describe("Scylla Manager integration", func() {
 		o.Expect(err).NotTo(o.HaveOccurred())
 
 		framework.By("Verifying that repair task deletion was synchronized")
-		tasks, err = managerClient.ListTasks(ctx, *sc.Status.ManagerID, "repair", false, "", "")
+		// Progressing conditions of the underlying ScyllaDBManagerTasks are not propagated to ScyllaClusters by the migration controller,
+		// neither does the controller await the deletion of ScyllaDBManagerTasks.
+		// Task deletion from ScyllaDB Manager state is asynchronous to ScyllaCluster state update.
+		o.Eventually(func(eo o.Gomega, ctx context.Context) {
+			tasks, err = managerClient.ListTasks(ctx, managerClusterID, "repair", false, "", "")
+			eo.Expect(err).NotTo(o.HaveOccurred())
+			eo.Expect(tasks.TaskListItemSlice).To(o.BeEmpty())
+		}).
+			WithContext(ctx).
+			WithTimeout(utils.SyncTimeout).
+			WithPolling(5 * time.Second).
+			Should(o.Succeed())
+
+		framework.By("Deleting ScyllaCluster")
+		err = f.ScyllaClient().ScyllaV1().ScyllaClusters(sc.Namespace).Delete(
+			ctx,
+			sc.Name,
+			metav1.DeleteOptions{
+				PropagationPolicy: pointer.Ptr(metav1.DeletePropagationForeground),
+				Preconditions: &metav1.Preconditions{
+					UID: &sc.UID,
+				},
+			})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Waiting for ScyllaCluster to be deleted")
+		deletionCtx, deletionCtxCancel := context.WithTimeout(ctx, utils.SyncTimeout)
+		defer deletionCtxCancel()
+		err = framework.WaitForObjectDeletion(
+			deletionCtx,
+			f.DynamicClient(),
+			scyllav1.GroupVersion.WithResource("scyllaclusters"),
+			sc.Namespace,
+			sc.Name,
+			&sc.UID,
+		)
 		o.Expect(err).NotTo(o.HaveOccurred())
-		o.Expect(tasks.TaskListItemSlice).To(o.BeEmpty())
+
+		framework.By("Verifying that the cluster was removed from the global ScyllaDB Manager state")
+		o.Eventually(func(eo o.Gomega, ctx context.Context) {
+			_, err := managerClient.GetCluster(ctx, managerClusterID)
+			eo.Expect(err).To(o.HaveOccurred())
+			eo.Expect(err).To(o.Satisfy(managerclienterrors.IsNotFound))
+		}).
```

Comment:
I can see we do not have a complementary check for verifying whether the cluster got registered in the manager in the first place, ergo we do not get 404  (we have a check relying on `IsScyllaClusterRegisteredWithManager` helper that reads `ScyllaCluster`'s status, but doesn't verify state on the manager side). Should we add one?

Reply:
Manager's ClusterID can only be propagated to ScyllaCluster's status if the cluster was registered with manager, so I'm not sure there's much benefit to it. Or would you do it just for confidence?